core/src/main/java/org/gbif/pipelines/core/ws/client/internal/DatasetMetaInfoServiceClient.java

```diff
@@ -0,0 +1,166 @@
+package org.gbif.pipelines.core.ws.client.internal;
+
+import org.gbif.pipelines.core.ws.HttpClientFactory;
+import org.gbif.pipelines.core.ws.HttpConfigFactory;
+import org.gbif.pipelines.core.ws.config.Config;
+import org.gbif.pipelines.core.ws.config.Service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import okhttp3.OkHttpClient;
+import retrofit2.Call;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * rest client for getting gbif internal api responses
+ */
+public class DatasetMetaInfoServiceClient {
+
+  private static final String INSTALLATION_KEY = "installationKey";
+  private static final String PUB_ORGANIZATION_KEY = "publishingOrganizationKey";
+  private static final String DATASET_TITLE_KEY = "title";
+  private static final String PUB_ORG_COUNTRY_KEY = "country";
+  private static final String INSTALLATION_TYPE_KEY = "type";
+  private static final String DATASET_NETWORK_KEY = "key";
+  private final DatasetMetaInfoService service;
+
+  /**
+   * create client from provided configuration
+   */
+  public static DatasetMetaInfoServiceClient from(Config wsConfig) {
+    // create client
+    OkHttpClient client = HttpClientFactory.createClient(wsConfig);
+
+    // create service
+    Retrofit retrofit = new Retrofit.Builder().client(client)
+      .baseUrl(wsConfig.getBasePath())
+      .addConverterFactory(GsonConverterFactory.create())
+      .validateEagerly(true)
+      .build();
+
+    return new DatasetMetaInfoServiceClient(retrofit.create(DatasetMetaInfoService.class));
+  }
+
+  /**
+   * initialize Client with default values
+   */
+  public static DatasetMetaInfoServiceClient client() {
+    return DatasetMetaInfoServiceClient.from(HttpConfigFactory.createConfig(Service.DATASET_META));
+  }
+
+  private DatasetMetaInfoServiceClient(DatasetMetaInfoService internalService) { this.service = internalService;}
+
+  /**
+   * perform webservice call when needed to aggregate the needed GBIF terms
+   *
+   * @param datasetUUID datasetUUID
+   *
+   * @return aggregated GBIFTerms response for the provided datasetUUID
+   */
+  public DatasetMetaInfoResponse getDatasetMetaInfo(String datasetUUID) throws ExecutionException {
```

Comment:
a minor suggestion:

 /**
   * Perform webservice call when needed to aggregate the needed GBIF terms
   *
   * @param datasetUUID datasetUUID
   *
   * @return aggregated GBIFTerms response for the provided datasetUUID
   */
  public DatasetMetaInfoResponse getDatasetMetaInfo(String datasetUUID) throws ExecutionException {
    Objects.requireNonNull(datasetUUID,"DatasetUUID cannot be null");

    DatasetMetaInfoResponse.Builder response = new DatasetMetaInfoResponse.Builder();
    response.withDatasetKey(datasetUUID);

    JsonObject dataset = getDataset(datasetUUID);

    consumeValue(dataset, DATASET_TITLE_KEY, response::withDatasetTitle);

    consumeValue(dataset, PUB_ORGANIZATION_KEY,
                 orgKey -> {
                    response.withPublishingOrgKey(orgKey);
                    consumeValue(getOrganization(orgKey), PUB_ORG_COUNTRY_KEY, response::withPublishingCountry);
                 });

    consumeValue(dataset, INSTALLATION_KEY,
                 installationKey -> consumeValue(getInstallation(installationKey), INSTALLATION_TYPE_KEY,
                                                 response::withProtocol));

    Optional.ofNullable(getNetworkFromDataset(datasetUUID))
            .map(networks ->
                    StreamSupport.stream(getNetworkFromDataset(datasetUUID).spliterator(), false)
                            .map(element -> element.getAsJsonObject().get(DATASET_NETWORK_KEY).getAsString())
                            .collect(Collectors.toList()))
            .ifPresent(response::withNetworkKey);

    return response.build();
  }

  /**
   * Utility method to consume optional primitive values form JsonNode.
   */
  private void consumeValue(JsonObject jsonObject, String simpleProperty, Consumer<String> consumer) {
      Optional.ofNullable(jsonObject.getAsJsonPrimitive(simpleProperty))
              .map(JsonPrimitive::getAsString)
              .ifPresent(consumer);
  }